x/auth/tx/decoder.go

```diff
@@ -79,3 +89,75 @@ func DefaultJSONTxDecoder(cdc codec.ProtoCodecMarshaler) sdk.TxDecoder {
 		}, nil
 	}
 }
+
+// rejectNonADR027 rejects txBytes that do not follow ADR-027. This function
+// only checks that:
+// - field numbers are in ascending order (1, 2, and potentially multiple 3s),
+// - and varints as as short as possible.
+// All other ADR-027 edge cases (e.g. TxRaw fields having default values) will
+// not happen with TxRaw.
+func rejectNonADR027(txBytes []byte) error {
+	// Make sure all fields are ordered in ascending order with this variable.
+	prevTagNum := protowire.Number(0)
+
+	for len(txBytes) > 0 {
+		tagNum, _, m := protowire.ConsumeTag(txBytes)
```

Comment:
Do we not need to check that the wire type is `2` length delimited? Maybe it's implicit but I don't think it hurts... Wdyt?

Reply:
I also thought an error would be thrown downstream if it's not `2`, but it's a cheap check, so I added.